### PR TITLE
IDA plugin now support arguments

### PR DESCRIPTION
**Pull request by zi0Black**
Added support for custom binary arguments, for IDA plugin inside qilingida.py.

Updated ChangeLog.

### CODE REVIEW

File: qiling/extensions/idaplugin/qilingida.py [hunk 1]
```diff
@@ -2059,11 +2061,16 @@ def ql_set_rootfs(self):
 
         rootfspath = setupdlg.path_name.value
         customscript = setupdlg.script_name.value
+        parameter = setupdlg.parameters.value
         env = setupdlg.env_var.value
 
         if customscript != '':
             self.customscriptpath = customscript
 
+        para_array = parameter.split(' ')
```
Comment (wtdcode) [inline qiling/extensions/idaplugin/qilingida.py:2070]: Need a more robust parser. e.g. parsing strings including spaces.

**Review by wtdcode — CHANGES_REQUESTED** [review]

File: qiling/extensions/idaplugin/qilingida.py (hunk 1 above)
Comment (zi0Black) [reply qiling/extensions/idaplugin/qilingida.py:2070]: Probably a regex or a proper grammar for ast could do the job, WDYT?

Comment (zi0Black) [reply qiling/extensions/idaplugin/qilingida.py:2070]: Or we can just demand everything from the user, and ask for a JSON as we did for the custom ENV.

Comment (wtdcode) [reply qiling/extensions/idaplugin/qilingida.py:2070]: Not sure. One case I'm considering is like:

`printf "Hello %s" "world!"`

Comment (wtdcode) [reply qiling/extensions/idaplugin/qilingida.py:2070]: A relevant post: https://stackoverflow.com/questions/1706551/parse-string-into-argv-argc

Comment (zi0Black) [reply qiling/extensions/idaplugin/qilingida.py:2070]: Just have a user-supplied JSON (from a file)?

e.x:
{"-c","aaaaa","bbbbb","\"c deefedede\""}

Comment (zi0Black) [reply qiling/extensions/idaplugin/qilingida.py:2070]: > A relevant post: https://stackoverflow.com/questions/1706551/parse-string-into-argv-argc

BTW, I will look at it, in the next few days.

**Comment by xwings** [comment]
please update the conflict too

File: qiling/extensions/idaplugin/qilingida.py (hunk 1 above)
Comment (zi0Black) [reply qiling/extensions/idaplugin/qilingida.py]: I'm now using `shlex` in order to parse arguments @wtdcode.
Input: `-c aaaa -j "aa bb"`

parsed: `['-c', 'aaaa', '-j', 'aa bb']`

**Comment by wtdcode** [comment]
@zi0Black Nice work, thanks!

**Pull request merged**